timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py

```diff
@@ -333,3 +333,38 @@ def predict(
 
     def _more_tags(self) -> dict:
         return {"can_refit_full": True}
+
+
+@dataclass(frozen=True)
+class MLFMemoryUsage:
+    """Estimated peak memory usage of MLForecast.preprocess (in MB) using the following linear model:
+
+    peak_mem_usage = a * num_rows * num_features + b * num_rows * c
+    """
+
+    a = 3.16e-5
+    b = 1.4e-4
+    c = 661
```

Comment:
Can we add test cases to ensure the longevity of this linear model despite updates to our model over time?